src/ray/protobuf/gcs_service.proto

```diff
@@ -431,12 +431,46 @@ message RegisterWorkerReply {
   GcsStatus status = 1;
 }
 
+message GetWorkerInfoRequest {
+  // ID of this worker.
+  bytes worker_id = 1;
+}
+
+message GetWorkerInfoReply {
+  GcsStatus status = 1;
+  // Data of worker.
+  WorkerTableData worker_table_data = 2;
+}
+
+message GetAllWorkerInfoRequest {
+}
+
+message GetAllWorkerInfoReply {
+  GcsStatus status = 1;
+  // Data of worker
+  repeated WorkerTableData worker_table_data = 2;
+}
+
+message AddWorkerInfoRequest {
+  WorkerTableData worker_data = 1;
+}
+
+message AddWorkerInfoReply {
+  GcsStatus status = 1;
+}
+
 // Service for worker info access.
 service WorkerInfoGcsService {
   // Report a worker failure to GCS Service.
   rpc ReportWorkerFailure(ReportWorkerFailureRequest) returns (ReportWorkerFailureReply);
   // Register a worker to GCS Service.
   rpc RegisterWorker(RegisterWorkerRequest) returns (RegisterWorkerReply);
+  // Get worker information from GCS Service by worker id.
+  rpc GetWorkerInfo(GetWorkerInfoRequest) returns (GetWorkerInfoReply);
+  // Get information of all workers from GCS Service.
+  rpc GetAllWorkerInfo(GetAllWorkerInfoRequest) returns (GetAllWorkerInfoReply);
+  // Add worker information to GCS Service.
+  rpc AddWorkerInfo(AddWorkerInfoRequest) returns (AddWorkerInfoReply);
```

Comment:
Is this duplicate with `RegisterWorkerRequest`? We can just reuse `RegisterWorkerRequest`.

Reply:
I try to reuse `RegisterWorkerRequest`, but it will cause an error in the macro `VOID_GCS_RPC_CLIENT_METHOD` in `gcs_rpc_client.h:52`